### PR TITLE
Initial check-in of R2R format documentation

### PR DESCRIPTION
**Pull request by trylek**
In this initial commit I have converted Jan Kotas' R2R format doc
from an internal Word document to the md format and I removed some
references to other internal docs. I plan to follow up by updating
the description to the current format version (e.g. apparently the
description doesn't yet include the instance entrypoint table) and
by adding specific extensions for the composite R2R format.

Thanks

Tomas

P.S. This is my first attempt at a md documentation PR so please
don't hesitate to steer my efforts if I happen to not be sticking
to any doc coding standards I might be unaware of.

### CODE REVIEW

**Comment by trylek** [comment]
OK, based on Jan's supportive review I have decided to remove the WIP modifier and merge this in separately as a well-defined initial checkpoint. I'm going to wait until tomorrow in case Michal and / or Simon decide to provide additional feedback - I don't want to waste energy formatting the md before I'm sure we're on the same page what the correct formatting should look like.

**Comment by trylek** [comment]
By the way, do we need to have something akin to source code copyright notices in design docs? I assume that patentable matter is not published on GitHub before actually patented so that should hopefully not be a concern.

File: Documentation/botr/readytorun-format.md [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+    READYTORUN_HELPER_CheckedWriteBarrier_EBP   = 0x10B,
+
+    // JIT32 x86-specific exception handling
+    READYTORUN_HELPER_EndCatch                  = 0x110,
+};
+```
+
+# References
+
+[1: Managed Executables with Native Code](https://microsoft.sharepoint.com/teams/netfx/CLR/coreclr/Shared%20Documents1/ReadyToRun/ManagedNativeCode.docx)
```
Comment (MichalStrehovsky) [inline Documentation/botr/readytorun-format.md]: This is an internal-only link.

Comment (trylek) [reply Documentation/botr/readytorun-format.md]: Thanks for pointing that out, removed in 2nd commit.

**Pull request merged**